src/api.js

```diff
@@ -782,6 +782,13 @@ API.prototype.declare = function(options, handler) {
     options.output = this._options.schemaPrefix + options.output;
   }
   this._entries.push(options);
+  var valueArr = options.map((item) => {return item.name;});
+  var isDuplicate = valueArr.some(function(item, idx) { 
+    return valueArr.indexOf(item) != idx; 
```

Comment:
You will want to see if there is an item in `this._entries` which has the name `options.name`.  You'll need to do that *before* `options` is pushed onto `this._entries`, though.